### ISSUE TITLE
Cannot use Poetry file-based dependencies

### ISSUE THREAD

**Issue by phillipuniverse** [dependabot/dependabot-core#4333]
I am almost certain that this is the same problem detailed in #1771. The difference is rather than using a path to a directory, I am using a path to a file. Here is the relevant part of my `pyproject.toml`:

```toml
[tool.poetry.dependencies]
geocoding = { file = "here_api/geocoding-7.65.tar.gz" }
```

This produces the following in `poetry.lock`:

```toml
[package.source]
type = "file"
url = "here_api/geocoding-7.65.tar.gz"
```

Dependabot errors with almost the exact same error as #1771:

```
pdater | ERROR <job_223616772> Error processing types-requests (Dependabot::SharedHelpers::HelperSubprocessFailed)
updater | ERROR <job_223616772> RuntimeError
updater | <job_223616772> 
updater | <job_223616772>   The Poetry configuration is invalid:
updater | <job_223616772>     - [dependencies.geocoding] {'file': 'here_api/geocoding-7.65.tar.gz', 'version': '7.65'} is not valid under any of the given schemas
updater | <job_223616772>   
updater | <job_223616772> 
updater | <job_223616772>   at /usr/local/.pyenv/versions/3.8.7/lib/python3.8/site-packages/poetry/core/factory.py:43 in create_poetry
updater | <job_223616772>        39│             message = ""
updater | <job_223616772>        40│             for error in check_result["errors"]:
updater | <job_223616772>        41│                 message += "  - {}\n".format(error)
updater | <job_223616772>        42│ 
updater | <job_223616772>     →  43│             raise RuntimeError("The Poetry configuration is invalid:\n" + message)
updater | <job_223616772>        44│ 
updater | <job_223616772>        45│         # Load package
updater | <job_223616772>        46│         name = local_config["name"]
updater | <job_223616772>        47│         version = local_config["version"]
```

I believe the fix is almost the exact same one-line fix applied in #2361 but also needs to special case `type=="file"`.

**Package ecosystem**
`pip`
**Package manager version**
Poetry 1.1.8
**Language version**
Python 3.8.7
**Manifest location and content prior to update**
poetry.lock, pyproject.toml

**dependabot.yml content**

```yaml
version: 2

registries:
  private-pypi:
    type: python-index
    url: https://pypi.myrepo.com
    username: user
    password: ${{secrets.PYPI_PASS}}

updates:
  - package-ecosystem: "pip"
    directory: "/"
    insecure-external-code-execution: allow
    registries:
      - private-pypi
    schedule:
      interval: daily
    reviewers:
      - "someteam"
```

**Updated dependency**
N/A
**What you expected to see, versus what you actually saw**
Poetry updating to work
**Native package manager behavior**
N/A, not exactly doing an update
**Images of the diff or a link to the PR, issue or logs**

**🕹 Bonus points: Smallest manifest that reproduces the issue**

**Comment by phillipuniverse** [comment]
Also -- I'm using the GitHub native Dependabot.

**Comment by phillipuniverse** [comment]
Looks like file-based dependencies are still not 100% fixed, new error at #4456.
